src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -50,28 +50,20 @@ public ForwardedHeadersMiddleware(RequestDelegate next, ILoggerFactory loggerFac
         ArgumentNullException.ThrowIfNull(options);
 
         // Make sure required options is not null or whitespace
-        EnsureOptionNotNullorWhitespace(options.Value.ForwardedForHeaderName, nameof(options.Value.ForwardedForHeaderName));
-        EnsureOptionNotNullorWhitespace(options.Value.ForwardedHostHeaderName, nameof(options.Value.ForwardedHostHeaderName));
-        EnsureOptionNotNullorWhitespace(options.Value.ForwardedProtoHeaderName, nameof(options.Value.ForwardedProtoHeaderName));
-        EnsureOptionNotNullorWhitespace(options.Value.ForwardedPrefixHeaderName, nameof(options.Value.ForwardedPrefixHeaderName));
-        EnsureOptionNotNullorWhitespace(options.Value.OriginalForHeaderName, nameof(options.Value.OriginalForHeaderName));
-        EnsureOptionNotNullorWhitespace(options.Value.OriginalHostHeaderName, nameof(options.Value.OriginalHostHeaderName));
-        EnsureOptionNotNullorWhitespace(options.Value.OriginalProtoHeaderName, nameof(options.Value.OriginalProtoHeaderName));
-        EnsureOptionNotNullorWhitespace(options.Value.OriginalPrefixHeaderName, nameof(options.Value.OriginalPrefixHeaderName));
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.ForwardedForHeaderName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.ForwardedHostHeaderName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.ForwardedProtoHeaderName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.ForwardedPrefixHeaderName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.OriginalForHeaderName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.OriginalHostHeaderName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.OriginalProtoHeaderName);
+        ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.OriginalPrefixHeaderName);
 
         _options = options.Value;
         _logger = loggerFactory.CreateLogger<ForwardedHeadersMiddleware>();
         _next = next;
 
         PreProcessHosts();
-
-        static void EnsureOptionNotNullorWhitespace(string value, string propertyName)
-        {
-            if (string.IsNullOrWhiteSpace(value))
-            {
-                throw new ArgumentException($"options.{propertyName} is required", nameof(options));
```

Comment:
Given that this is thrown form a constructor that has non-options parameters, the old message was far clearer because at least it mentions "options".

Reply:
With `ArgumentException.ThrowIfNullOrWhiteSpace(options.Value.ForwardedForHeaderName)`, the entire expression is captured as parameter name and "options" is mentioned: 

```
System.ArgumentNullException: Value cannot be null. (Parameter 'options.Value.ForwardedForHeaderName')
     at System.ArgumentNullException.Throw(String paramName)
     at System.ArgumentNullException.ThrowIfNull(Object argument, String paramName)
     at System.ArgumentException.ThrowNullOrWhiteSpaceException(String argument, String paramName)
     at System.ArgumentException.ThrowIfNullOrWhiteSpace(String argument, String paramName)
     at ForwardedHeadersMiddleware..ctor(RequestDelegate next, ILoggerFactory loggerFactory, IOptions`1 options)
```

Do you still want me to revert this change?